pydust/src/pytypes.zig

```diff
@@ -599,6 +599,40 @@ fn BinaryOperator(
     };
 }
 
+fn EqualsOperator(
+    comptime definition: type,
+    comptime op: []const u8,
+) type {
+    return struct {
+        const equals = std.mem.eql(u8, op, "__eq__");
+        fn call(pyself: *ffi.PyObject, pyother: *ffi.PyObject) callconv(.C) ?*ffi.PyObject {
+            const func = @field(definition, op);
+            const typeInfo = @typeInfo(@TypeOf(func));
+            const sig = funcs.parseSignature(op, typeInfo.Fn, &.{});
+
+            if (sig.selfParam == null) @compileError(op ++ " must take a self parameter");
+            if (sig.nargs != 1) @compileError(op ++ " must take exactly one parameter after self parameter");
+
+            // If other arg is of the same type as self we can short circuit equality of both objects aren't of same type
+            if (sig.argsParam == *definition or sig.argsParam == *const definition) {
```

Comment:
If you mutate `other` in the __eq__ function then you don't deserve nice things!